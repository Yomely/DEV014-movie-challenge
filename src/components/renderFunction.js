/* eslint-disable*/
export const renderItems = (data) => {
  const ul = document.createElement('ul');
  ul.classList.add('flex-container');

  data.forEach((movie) => {
    const movieLi = document.createElement('li');
    // paso 1 
    // agregarle un evento al darle clicl al li
    /// console.log e imprimir el id de la pelicula
    // paso 3 es invocar la funcion navigateTo para navegar al detalle de la pelicula


    movieLi.setAttribute('itemscope', '');
    movieLi.setAttribute('itemtype', movie.id);

    const semanticHTML = `
      <dl itemscope itemtype="MovieChallenge">
      <img src="https://image.tmdb.org/t/p/w500${movie.poster_path}" alt="${movie.title}" />
      <dd itemprop="name">${movie.title}</dd>
      <dd itemprop="releaseDate">${movie.release_date}</dd>
      </dl>  
       `;
    movieLi.innerHTML = semanticHTML;
    movieLi.classList.add('flex-item');

    ul.append(movieLi);
  });
  return ul;
};
