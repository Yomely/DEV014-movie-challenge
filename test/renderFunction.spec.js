/* eslint-disable no-undef */
import { renderItems } from '../src/components/renderFunction.js';
import fakeData from './mockData.js';

describe('renderItems', () => {
  let root;
  beforeEach(() => {
    document.body.innerHTML += '<div id="app"> </div>';
    root = document.querySelector('#app');
    root.appendChild(renderItems(fakeData));
  });

  // sobre  root escribir una prueba que combruebe que hay 3 li en el componente
  // ecribir tu test
  // hacer una variable que tennga un query selector all
  // hacer una asercion de que su logitud sea 3
  // Matchers, al proposito de los test, el patron AAA en test

  it('Should show there are 3 movies like length', () => {
    const liElements = document.querySelectorAll('li');
    expect(liElements.length).toBe(3);
  });

  it('Should take a picture 3 movies', () => {
    expect(root).toMatchSnapshot();
  });
});
