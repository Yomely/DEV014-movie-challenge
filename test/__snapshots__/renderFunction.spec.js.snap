// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderItems Sould take a picture 3 movies 1`] = `
<div
  id="app"
>
   
  <ul
    class="flex-container"
  >
    <li
      class="flex-item"
      itemscope=""
      itemtype="Lalaland"
    >
      
      
      <dl
        itemscope=""
        itemtype="MovieChallenge"
      >
        
      
        <img
          alt="lalaland"
          src="https://image.tmdb.org/t/p/w50012312.png"
        />
        
      
        <dd
          itemprop="name"
        >
          lalaland
        </dd>
        
      
        <dd
          itemprop="releaseDate"
        >
          2022
        </dd>
        
      
      </dl>
        
       
    </li>
    <li
      class="flex-item"
      itemscope=""
      itemtype="Toreto"
    >
      
      
      <dl
        itemscope=""
        itemtype="MovieChallenge"
      >
        
      
        <img
          alt="toreto"
          src="https://image.tmdb.org/t/p/w5004535.png"
        />
        
      
        <dd
          itemprop="name"
        >
          toreto
        </dd>
        
      
        <dd
          itemprop="releaseDate"
        >
          2018
        </dd>
        
      
      </dl>
        
       
    </li>
    <li
      class="flex-item"
      itemscope=""
      itemtype="rango"
    >
      
      
      <dl
        itemscope=""
        itemtype="MovieChallenge"
      >
        
      
        <img
          alt="rango"
          src="https://image.tmdb.org/t/p/w5009o89p.png"
        />
        
      
        <dd
          itemprop="name"
        >
          rango
        </dd>
        
      
        <dd
          itemprop="releaseDate"
        >
          2006
        </dd>
        
      
      </dl>
        
       
    </li>
  </ul>
  <ul
    class="flex-container"
  >
    <li
      class="flex-item"
      itemscope=""
      itemtype="Lalaland"
    >
      
      
      <dl
        itemscope=""
        itemtype="MovieChallenge"
      >
        
      
        <img
          alt="lalaland"
          src="https://image.tmdb.org/t/p/w50012312.png"
        />
        
      
        <dd
          itemprop="name"
        >
          lalaland
        </dd>
        
      
        <dd
          itemprop="releaseDate"
        >
          2022
        </dd>
        
      
      </dl>
        
       
    </li>
    <li
      class="flex-item"
      itemscope=""
      itemtype="Toreto"
    >
      
      
      <dl
        itemscope=""
        itemtype="MovieChallenge"
      >
        
      
        <img
          alt="toreto"
          src="https://image.tmdb.org/t/p/w5004535.png"
        />
        
      
        <dd
          itemprop="name"
        >
          toreto
        </dd>
        
      
        <dd
          itemprop="releaseDate"
        >
          2018
        </dd>
        
      
      </dl>
        
       
    </li>
    <li
      class="flex-item"
      itemscope=""
      itemtype="rango"
    >
      
      
      <dl
        itemscope=""
        itemtype="MovieChallenge"
      >
        
      
        <img
          alt="rango"
          src="https://image.tmdb.org/t/p/w5009o89p.png"
        />
        
      
        <dd
          itemprop="name"
        >
          rango
        </dd>
        
      
        <dd
          itemprop="releaseDate"
        >
          2006
        </dd>
        
      
      </dl>
        
       
    </li>
  </ul>
</div>
`;
